pkg/node/volumemgr.go

```diff
@@ -886,56 +889,60 @@ func (m *VolumeManager) discoverLVGOnSystemDrive() error {
 		}
 	}
 
-	var (
-		rootMountPoint, vgName string
-		vgFreeSpace            int64
-	)
-
-	if rootMountPoint, err = m.fsOps.FindMountPoint(base.KubeletRootPath); err != nil {
-		ll.Errorf("Failed to find root mountpoint for %s, error: %v, try to find for %s",
-			base.KubeletRootPath, err, base.HostRootPath)
-		if rootMountPoint, err = m.fsOps.FindMountPoint(base.HostRootPath); err != nil {
-			return fmt.Errorf(errTmpl, err)
-		}
+	// 2. check whether there is LVG configuration on the system drive or not
+	var driveCR = new(drivecrd.Drive)
+	// TODO: handle situation when there is more then one system drive
+	if err = m.k8sClient.ReadCR(context.Background(), m.systemDrivesUUIDs[0], driveCR); err != nil {
+		return err
 	}
 
-	// from container we expect here name like "VG_NAME[/var/lib/kubelet/pods]"
-	rootMountPoint = strings.Split(rootMountPoint, "[")[0]
-
-	devices, err := m.listBlk.GetBlockDevices(rootMountPoint)
+	pvs, err := m.lvmOps.GetAllPVs()
 	if err != nil {
-		return fmt.Errorf(errTmpl, err)
+		return fmt.Errorf("unable to list PVs on the system: %v", err)
 	}
 
-	if devices[0].Rota != base.NonRotationalNum {
+	var systemPVName string
+	for _, pv := range pvs {
+		if strings.Contains(pv, driveCR.Spec.Path) {
+			systemPVName = pv
+			break
+		}
+	}
+
+	if systemPVName == "" {
+		ll.Info("There is no LVM configuration on the system drive")
 		m.discoverLvgSSD = false
-		ll.Infof("System disk is not SSD. LVG will not be created base on it.")
 		return nil
 	}
 
-	lvgExists, err := m.lvmOps.IsLVGExists(rootMountPoint)
-
+	// 3. check whether the system drive SSD or not
+	devices, err := m.listBlk.GetBlockDevices(driveCR.Spec.Path)
 	if err != nil {
-		return fmt.Errorf(errTmpl, err)
+		return fmt.Errorf("unable to get info about system drive: %s", err)
 	}
 
-	if !lvgExists {
+	if devices[0].Rota != base.NonRotationalNum {
```

Comment:
We already renamed our SC from system-ssdlvg to syslvg. Maybe we should remove this check and support system VG on all storage types?

Reply:
good point, going to refactor...

Reply:
refactored